sentence_transformers/SentenceTransformer.py

```diff
@@ -572,11 +578,48 @@ def get_sentence_features(self, *features):
         return self._first_module().get_sentence_features(*features)
 
     def get_sentence_embedding_dimension(self):
+        """
+        :return: The number of dimensions in the output of `encode`. If it's not known, it's `None`.
+        """
+        output_dim = None
         for mod in reversed(self._modules.values()):
             sent_embedding_dim_method = getattr(mod, "get_sentence_embedding_dimension", None)
             if callable(sent_embedding_dim_method):
-                return sent_embedding_dim_method()
-        return None
```

Comment:
I'm not sure what causes `return None`, i.e., when there's never a `get_sentence_embedding_dimension` attribute in the modules. In this case, the

```python
return min(output_dim or np.inf, self.output_dim)
```

returns `self.output_dim`, which could be greater than whatever dimension actually ends up being outputted.

Reply:
The `return None` is kind of moot, in practice this will essentially never happen: only if the `SentenceTransformer` is created with incorrect Modules. I think your current setup should be totally fine, at a glance.